### PR TITLE
feat: Adding embedding

### LINKED ISSUE #182: Add support for embedding

### PR DESCRIPTION
**Pull request by nickrandolph**
GitHub Issue (If applicable): closes #182 

## PR Type

What kind of change does this PR introduce?
- Feature

## What is the new behavior?

Template includes embedding sample

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences. Validate PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.Templates/content/unoapp/MyExtensionsApp._1/App.blank.cs [hunk 1]
```diff
@@ -13,6 +18,13 @@ protected override void OnLaunched(LaunchActivatedEventArgs args)
 		MainWindow = Microsoft.UI.Xaml.Window.Current;
 #endif
 
+//+:cnd:noEmit
+#if mauiEmbedding
+		Uno.Extensions.Maui.MauiEmbedding.UseMauiEmbedding(this, maui => maui
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp._1/App.blank.cs:23]: This needs to be fixed to be `this.UseMauiEmbedding`

File: src/Uno.Templates/content/unoapp/MyExtensionsApp._1/MainPage.xaml [hunk 2]
```diff
@@ -10,6 +10,13 @@
 	<StackPanel HorizontalAlignment="Center"
 				VerticalAlignment="Center">
 		<TextBlock AutomationProperties.AutomationId="HelloTextBlock"
-				   Text="Hello Uno Platform" />
+				   	Text="Hello Uno Platform" 
+				   	HorizontalAlignment="Center" />
+<!--#if (mauiEmbedding)-->
+		<embed:MauiHost x:Name="MauiHostElement"
+					xmlns:embed="using:Uno.Extensions.Maui">
+					<controls:EmbeddedControl xmlns:controls="using:MyExtensionsApp._1.MauiControls" />
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp._1/MainPage.xaml:18]: @dansiegel not sure if you want to move namespaces out to the top of the file??

File: src/Uno.Templates/content/unoapp/MyExtensionsApp._1/Presentation/MainPage.xaml [hunk 3]
```diff
@@ -24,6 +24,12 @@
 					HorizontalAlignment="Center"
 					VerticalAlignment="Center"
 					Spacing="16">
+<!--#if (mauiEmbedding)-->
+		<embed:MauiHost x:Name="MauiHostElement"
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp._1/Presentation/MainPage.xaml:28]: @dansiegel as per previous comment - do you want to move namespaces to top of file?

**Review by dansiegel — APPROVED** [review]

File: src/Uno.Templates/content/unoapp/.template.config/template.json [hunk 4]
```diff
@@ -749,6 +757,11 @@
       "datatype": "bool",
       "defaultValue": "false"
     },
+    "useMauiPackageReference": {
+      "type": "computed",
+      "datatype": "bool",
+      "value": "(mauiEmbedding && tfm != 'net7.0')"
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/.template.config/template.json:763]:
```suggestion
      "value": "(mauiEmbedding && tfm == 'net8.0')"
```

File: src/Uno.Templates/content/unoapp/MyExtensionsApp._1.Mobile/MyExtensionsApp._1.Mobile.csproj [hunk 5]
```diff
@@ -24,6 +24,11 @@
 		<!-- Required for C# Hot Reload -->
 		<UseInterpreter Condition="'$(Configuration)' == 'Debug' and $([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) != 'maccatalyst'">True</UseInterpreter>
 		<IsUnoHead>true</IsUnoHead>
+
+		<!--#if (mauiEmbedding)-->
+		<!-- Hot restart not working well with Maui Embedding -->
+		<UnoDisableHotRestartHelperGeneration>true</UnoDisableHotRestartHelperGeneration>
```
Comment (nickrandolph) [inline src/Uno.Templates/content/unoapp/MyExtensionsApp._1.Mobile/MyExtensionsApp._1.Mobile.csproj:30]: Need to recheck this - I suspect this isn't required any more as the source generators check for reference to maui

File: build/.azure-pipelines.TemplateValidation.yml [hunk 6]
```diff
@@ -67,6 +67,10 @@ jobs:
         templateArgs: '-preset=recommended -auth=msal'
       MsalAuthMvvm:
         templateArgs: '-preset=recommended -presentation mvvm -auth=msal'
+      #MauiBlankApp:
```
Comment (nickrandolph) [inline build/.azure-pipelines.TemplateValidation.yml:70]: These should be enabled

Comment (nickrandolph) [reply build/.azure-pipelines.TemplateValidation.yml:70]: https://github.com/unoplatform/uno.templates/issues/192

**Comment by nickrandolph** [comment]
@mergifyio backport legacy/4x

**Pull request merged**